api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -0,0 +1,257 @@
+      if (term instanceof UnboundTransform) {
+        return ((UnboundTransform<?, ?>) term).transform() + "(" + termToString(term.ref()) + ")";
+      } else if (term instanceof NamedReference) {
+        return ((NamedReference<?>) term).name();
+      } else {
+        throw new UnsupportedOperationException("Unsupported term: " + term);
+      }
+    }
+
+    @Override
+    public <T> String predicate(UnboundPredicate<T> pred) {
+      String term = termToString(pred.term());
+      switch (pred.op()) {
+        case IS_NULL:
+          return term + " IS NULL";
+        case NOT_NULL:
+          return term + " IS NOT NULL";
+        case IS_NAN:
+          return "is_nan(" + term + ")";
+        case NOT_NAN:
+          return "not_nan(" + term + ")";
+        case LT:
+          return term + " < " + sanitize(pred.literal());
+        case LT_EQ:
+          return term + " <= " + sanitize(pred.literal());
+        case GT:
+          return term + " > " + sanitize(pred.literal());
+        case GT_EQ:
+          return term + " >= " + sanitize(pred.literal());
+        case EQ:
+          return term + " = " + sanitize(pred.literal());
+        case NOT_EQ:
+          return term + " != " + sanitize(pred.literal());
+        case IN:
+          return term + " IN " + pred.literals().stream()
+              .map(ExpressionUtil::sanitize)
+              .collect(Collectors.joining(", ", "(", ")"));
+        case NOT_IN:
+          return term + " NOT IN " + pred.literals().stream()
+              .map(ExpressionUtil::sanitize)
+              .collect(Collectors.joining(", ", "(", ")"));
+        case STARTS_WITH:
+          return term + " STARTS WITH " + sanitize(pred.literal());
+        case NOT_STARTS_WITH:
+          return term + " NOT STARTS WITH " + sanitize(pred.literal());
+        default:
+          throw new UnsupportedOperationException("Cannot sanitize unsupported predicate type: " + pred.op());
+      }
+    }
+  }
+
+  private static String sanitize(Literal<?> literal) {
+    if (literal instanceof Literals.StringLiteral) {
+      CharSequence value = ((Literals.StringLiteral) literal).value();
+      if (DATE.matcher(value).matches()) {
+        return "(date)";
+      } else if (TIME.matcher(value).matches()) {
+        return "(time)";
+      } else if (TIMESTAMP.matcher(value).matches()) {
+        return "(timestamp)";
+      } else {
+        return sanitizeString(value);
+      }
+    } else if (literal instanceof Literals.DateLiteral) {
+      return "(date)";
+    } else if (literal instanceof Literals.TimeLiteral) {
+      return "(time)";
+    } else if (literal instanceof Literals.TimestampLiteral) {
+      return "(timestamp)";
+    } else if (literal instanceof Literals.IntegerLiteral) {
+      return sanitizeNumber(((Literals.IntegerLiteral) literal).value(), "int");
+    } else if (literal instanceof Literals.LongLiteral) {
+      return sanitizeNumber(((Literals.LongLiteral) literal).value(), "int");
```

Comment:
why not long?  (and below, why not double?)

Reply:
Good catch! I used `int` or `float` in both cases because the type is determined by what you pass in to create the literal, not the actual type of data in a table (because the expression is unbound). Since the types are floating anyway, I didn't think it made much sense to be strict. What really matters is the magnitude of the value. If you have a `4-digit-int` it doesn't matter if it was passed in as a long or passed in as an int.

Reply:
I see, though is there any disadvantage to just putting the exact one being passed in ?  I took a look and seems Spark does some schema-based binding of expression before passing it to Iceberg, so a comparison of long column results in long literal being passed down to Iceberg scan.   Though understand its not a huge point.

Reply:
I think that would require more processing later to use these sanitized values. Right now, this sanitizes log messages, but if you wanted to start collecting sanitized filters and doing analysis on them, one of the first steps would be to collapse these, I think. That's why I went ahead and normalized.